File: risk-lib/src/main/java/net/mjduffin/risk/lib/usecase/request/Request.java
package net.mjduffin.risk.lib.usecase.request;

public class Request {
    public enum Type {
        DRAFT,
        ATTACK,
        ENDATTACK,
        MOVE,
        FORTIFY,
        SKIPFORTIFY
    }

    protected Type requestType;

    public Type getRequestType() {
        return requestType;
    }
}

File: risk-lib/src/main/kotlin/net/mjduffin/risk/lib/usecase/request/Request.kt
package net.mjduffin.risk.lib.usecase.request

interface Request

data class MoveRequest(val playerName: String, val units: Int) : Request

data class FortifyRequest(
    val playerName: String,
    val fromTerritory: String,
    val toTerritory: String,
    val units: Int) : Request

data class EndAttackRequest(val playerName: String) : Request

data class DraftRequest(val player: String, val territory: String, val units: Int) : Request

data class AttackRequest(val player: String, val attacker: String, val defender: String) : Request

class SkipFortifyRequest : Request
